src/CachePeers.cc

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+
+#include "CachePeers.h"
+#include "SquidConfig.h"
+
+CachePeers::Storage::const_iterator
+CachePeers::nextPeerToPing()
+{
+    Assure(size());
+    const auto pos = peersPinged_ % size();
+    ++peersPinged_;
+    return storage.begin() + pos;
+}
+
+void
+CachePeers::remove(CachePeer *p)
+{
+    for (auto it = storage.begin(); it != storage.end(); ++it) {
+        if (it->get() == p) {
```

Comment:
If you have not already, please check that we cannot use, in the order of preference:

* std::find(..., p) + storage.erase(pos) instead of this explicit loop (which also hides an important "not found" clause better)
* `*it == p`